packages/protocol-http/src/Field.ts

```diff
@@ -0,0 +1,59 @@
+import { FieldPosition } from "./FieldPosition";
+
+export type FieldOptions = { 
+  name: string;
+  values?: string[];
+  kind?: FieldPosition
+};
+
+/**
+ * A name-value pair representing a single field
+ * transmitted in an HTTP Request or Response.
+ * 
+ * The kind will dictate metadata placement within
+ * an HTTP message.
+ * 
+ * All field names are case insensitive and
+ * case-variance must be treated as equivalent.
+ * Names MAY be normalized but SHOULD be preserved
+ * for accuracy during transmission.
+ */
+export class Field {
+  public name: string;
+  public values: string[];
+  public kind: FieldPosition;
```

Comment:
are these public fields meant to be readonly?

Reply:
Updated `kind` and `name` to be readonly